src/LobbyServer.hs

```diff
@@ -87,13 +87,13 @@ disconnectPlayerFromGame remoteGames remoteClientList sid = do
     Nothing -> return ()
     Just clientEntry -> liftIO $ CC.modifyMVar_ mVarGames $ \games -> mapM removePlayer games
       where
-        removePlayer (uuid, GameData ps gameName) =
-          let newClientList = filter ((name clientEntry /=) . name) ps in
-          return (uuid, GameData newClientList gameName)
+        removePlayer (uuid, gameData) =
+          let newClientList = filter ((name clientEntry /=) . name) $ players gameData in
+          return (uuid, gameData {players = newClientList})
 
 -- |Creates a new game on the server
-createGame :: Server GamesList -> Server ConcurrentClientList -> Server (Maybe String)
-createGame remoteGames remoteClientList = do
+createGame :: Server GamesList -> Server ConcurrentClientList -> Int -> Server (Maybe String)
```

Comment:
Here we already have a comment. But it tells us the same thing the function name does.  Comments for what the parameters represent would be great.